### ISSUE TITLE
Implement U2F authentication option

### ISSUE THREAD

**Issue by itismadness** [OPSnet/Gazelle#32]

**Comment by pjc09h** [comment]
FYI, BioGazelle has a robust WebAuthn/FIDO2 implementation. I'll link the [backend library documentation](https://webauthn-doc.spomky-labs.com/pure-php/the-hard-way), the [frontend library documentation](https://simplewebauthn.dev/docs/packages/browser), and the source for the various files:

- [Gazelle\WebAuthn\Base](https://github.com/biotorrents/gazelle/blob/development/app/WebAuthn/Base.php)
- [Gazelle\WebAuthn\CredentialSourceRepository](https://github.com/biotorrents/gazelle/blob/development/app/WebAuthn/CredentialSourceRepository.php)
- [Gazelle\WebAuthn\UserEntityRepository](https://github.com/biotorrents/gazelle/blob/development/app/WebAuthn/UserEntityRepository.php)
- [webAuthnAssert.js](https://github.com/biotorrents/gazelle/blob/development/resources/js/webAuthnAssert.js)
- [webAuthnCreate.js](https://github.com/biotorrents/gazelle/blob/development/resources/js/webAuthnCreate.js)
- [Database migration for credentials](https://github.com/biotorrents/gazelle/blob/development/database/migrations/20230513020218_web_authn.php)

Basically, the main components are the base class (common implementation) and a couple of small classes to store and retrieve credentials in the database (specific implementations). Then there's some JavaScript that hits your API to interact with the backend classes, do the handshakes, and communicate the results to the user.

Depending on where you decide to call `webAuthnAssert.js` in your login flow, you can use WebAuthn either as passwordless single-factor authentication or as multi-factor authentication with or without TOTP. I'm using the passwordless option to limit logins to one request.
